parsers/corelight-sensor.yaml

```diff
@@ -224,101 +657,86 @@ script: |
-          | network.direction := "external";
+      Vendor.local_orig=true AND Vendor.local_resp=true | network.direction := "internal";
+      Vendor.local_orig=true AND Vendor.local_resp=false | network.direction := "outbound";
+      Vendor.local_orig=false AND Vendor.local_resp=true | network.direction := "inbound";
+      Vendor.local_orig=false AND Vendor.local_resp=false | network.direction := "external";
       *
   }
 
-  | case {
-      _path = "ssl"
-          | network.transport := "ssl";
-      _path = "ssh"
-          | network.transport := "ssh";
-      _path = "smb_files"
-          | network.transport := "smb";
-      _path = "dns"
-          | network.transport := "dns";
-      _path = "http"
-          | network.transport := "http";
-      _path = "rdp"
-          | network.transport := "rdp";
-      *
+  | _path match {
+    "ssl" => network.transport := "ssl";
+    "ssh" => network.transport := "ssh";
+    "smb_files" => network.transport := "smb";
+    "dns" => network.transport := "dns";
+    "http" => network.transport := "http";
+    "rdp" => network.transport := "rdp";
+    * => *
   }
+
   | case {
-      Vendor.community_id = *
-          | network.community_id := rename(Vendor.community_id);
-      *
-          | communityId(as="network.community_id", proto=network.transport, sourceip=source.ip, sourceport=source.port, destinationip=destination.ip, destinationport=destination.port);
+      Vendor.community_id = * | network.community_id := Vendor.community_id;
+      * | communityId(as="network.community_id", proto=network.transport, sourceip=source.ip, sourceport=source.port, destinationip=destination.ip, destinationport=destination.port);
   }
   // in case there are no src/dst ip's or invalid ip's
   | case { network.community_id = "" | drop([network.community_id]); * }
 
   // VLAN Fields
-  | network.vlan.id := rename(Vendor.id.vlan)
-  | network.vlan.id := rename(Vendor.vlan)
+  | network.vlan.id := Vendor.id.vlan
+  | network.vlan.id := Vendor.vlan
 
   // File Fields
-  | file.size := rename(Vendor.total_bytes)
+  | file.size := Vendor.total_bytes
 
   // Hash Fields
-  | file.hash.md5 := rename(Vendor.md5)
-  | file.hash.sha256 := rename(Vendor.sha256)
-  | lower(file.hash.sha256, as="file.hash.sha256")
-  | lower(file.hash.md5, as="file.hash.md5")
+  | file.hash.md5 := lower(Vendor.md5)
+  | file.hash.sha256 := lower(Vendor.sha256)
 
   // HTTP Fields
-  | http.request.method := rename(Vendor.method)
-  | url.path := rename(Vendor.uri)
-  | url.domain := rename(Vendor.host)
-  | http.response.status_code := rename(Vendor.status_code)
+  | http.request.method := Vendor.method
+  | url.path := Vendor.uri
+  | url.domain := Vendor.host
+  | http.response.status_code := Vendor.status_code
   | case {
-  // Manually parse out mime response type
-      _path = "dns" AND Vendor.resp_mime_types[1] = * | @rawstring = /resp_mime_types":\[(?<http.response.mime_type>.*?)\]/;
```

Comment:
from CRWD
- regexing rawstring might be expensive here. Is there a way to point directly to specific fields?